libs/core/langchain_core/utils/function_calling.py

```diff
@@ -187,6 +189,81 @@ def convert_python_function_to_openai_function(
     )
 
 
+def _convert_typed_dict_to_openai_function(typed_dict: Type) -> FunctionDescription:
+    visited: Dict = {}
+    model = cast(
+        Type[BaseModel],
+        _convert_any_typed_dicts_to_pydantic(typed_dict, visited=visited),
+    )
+    return convert_pydantic_to_openai_function(model)
+
+
+_MAX_TYPED_DICT_RECURSION = 25
+
+
+def _convert_any_typed_dicts_to_pydantic(
+    type_: Type,
+    *,
+    visited: Dict,
+    depth: int = 0,
+) -> Type:
+    if type_ in visited:
+        return visited[type_]
+    elif depth >= _MAX_TYPED_DICT_RECURSION:
+        return type_
+    elif is_typeddict(type_):
```

Comment:
Does this work with typeddict from typing and from typing_extensions?

Reply:
yep, in unit tests